### PR TITLE
Windows project

### PR DESCRIPTION
**Pull request by EgorOrachyov**
WIP:

- [x] #5 Partially  
- [x] #8  
- [x] #9 
- [x] #10 
- [x] #11

### CODE REVIEW

**Comment by EgorOrachyov** [comment]
Current state: 
image

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+if(APPLE)
+    add_subdirectory(macos)
+    message(STATUS "Configure macos platform module")
+endif()
+
+if(WIN32)
+    add_subdirectory(windows)
+    message(STATUS "Configure windows platform module")
+endif()
```
Comment (tonsky) [inline CMakeLists.txt:16]: I guess we need Linux here?

File: shared/java/App.java [hunk 2]
```diff
@@ -25,6 +25,8 @@ public static Window makeWindow() {
         Class cls;
         if (Platform.CURRENT == Platform.MACOS)
             cls = App.class.forName("org.jetbrains.jwm.WindowMac");
+        else if (Platform.CURRENT == Platform.WINDOWS)
+            cls = App.class.forName("org.jetbrains.jwm.WindowWin32");
```
Comment (tonsky) [inline shared/java/App.java:29]: Also needs Linux? (might be a matter of rebase)

File: windows/cc/PlatformWin32.hh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <cwctype>
+#include <windows.h>
+#include <winuser.h>
+#include <shellscalingapi.h>
+
+#ifndef JWM_TRUE
```
Comment (tonsky) [inline windows/cc/PlatformWin32.hh:8]: Why not just use bool true/false?

**Review by tonsky — CHANGES_REQUESTED** [review]

**Review by tonsky — APPROVED** [review]

**Pull request merged**